docs/editor/extension-runtime-security.md

```diff
@@ -0,0 +1,82 @@
+PageTitle: Visual Studio Code Extension Runtime Security
+DateApproved: 12/11/2024
+MetaDescription: Learn about the security measures in place for Visual Studio Code extensions, including permissions, user reliability checks, and Marketplace protections.
+---
+
+# Extension runtime security
+
+[Extensions](/docs/editor/extension-marketplace.md) greatly enhance the functionality of Visual Studio Code. They can also introduce risks, such as malicious code execution and data privacy concerns. The [Visual Studio Marketplace](https://marketplace.visualstudio.com/vscode) has many ways to protect you from bad extensions. In addition, VS Code gives you several indicators of an extension's reliability.
+
+This document outlines the runtime permissions of extensions in VS Code and the measures in place to protect you from malicious extensions. You'll learn how to make an informed decision about the reliability of an extension before installing it.
+
+## About extension runtime permissions
+
+The [extension host](/api/advanced-topics/extension-host.md) is responsible for running extensions in VS Code. The extension host has the same permissions as VS Code itself. This means that any action that VS Code can perform, an extension can also perform through the extension host.
+
+For example, an extension can read and write files on your machine, make network requests, run external processes, and modify workspace settings.
+
+## Extension publisher trust
+
+As of VS Code release 1.97, when you first install an extension from a third-party publisher, VS Code shows a dialog prompting you to confirm that you trust the publisher of that extension.
+
+The list of trusted publishers is maintained in the `setting(extensions.trustedPublishers)` user setting.
+
+Publishers for extensions that you installed previously are considered trusted and are automatically added to the list of trusted publishers. You can also manually edit the list of trusted publishers.
+
+## Determine extension reliability
+
+Before you install an extension, you can take various steps to determine if it's reliable. The Visual Studio Marketplace provides you with information about the extension to help you make an informed decision:
+
+* **Ratings & Reviews**: Read what others think about the extension.
+
+* **Q & A**: Review existing questions and the level of the publisher's responsiveness. You can also engage with the extension's publisher if you have concerns.
+
+* **Issues, Repository, and License**: Check if the publisher provided these and if they have the support you expect.
+
+* **Verified Publisher**: Use the blue check mark next to the publisher's name and domain name as an extra signal of trust. The check mark indicates that the publisher has proven domain-name ownership to the Marketplace. It also shows that the Marketplace has verified both the existence of the domain name and the good standing of the publisher on the Marketplace for at least six months.
```

Comment:
Let's add a bullet point about "long-standing presence"
The goal is that publishers that have been publishing extensions for a long time should be more trust-worthy

Reply:
We don't have clear information about publisher's long-standing presence, apart from going through the history of a specific extension's releases. Am I missing a specific signal?

Reply:
Good point
Marketplace should update this page https://marketplace.visualstudio.com/publishers/vadimcn to contain more publisher information. I will discuss with @seaniyer @mariaghiondea 
The dialog points to that page.

Reply:
Though I still think we can add that point, and users can investigate when was the extension first published.
@ntrogh I leave that decision up to you nick